packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -52,6 +52,33 @@ import 'typography.dart';
 
 export 'package:flutter/services.dart' show Brightness;
 
+/// An interface that defines custom additions to a [ThemeData] object.
+///
+/// Typically used for custom colors. To use, subclass [ThemeExtension],
+/// define a number of fields (e.g. [Color]s), and implement the [copyWith] and
+/// [lerp] methods. The latter will ensure smooth transitions of properties when
+/// switching themes.
+///
+/// {@tool dartpad}
+/// This sample shows how to create and use a subclass of [ThemeExtension] that
+/// defines two colors.
+///
+/// ** See code in examples/api/lib/material/theme/theme_extension.1.dart **
+/// {@end-tool}
+abstract class ThemeExtension<T> {
```

Comment:
To get a little more runtime checking
abstract class ThemeExtension<T extends ThemeExtension<T>> 

This will ensure that 
class Something extends ThemeExtension<Other> {
}
Is invalid.

Reply:
Do you mean `abstract class ThemeExtension<T extends ThemeExtension<T>>`?

Doing so only works if all instances of `Map<Object, ThemeExtension<Object>>?` are replaced with `Map<Object, ThemeExtension<dynamic>>?` I'm not sure if there's a hidden downside to using `dynamic` as opposed to `Object`

Reply:
It can be Map<Object, ThemeExtension<ThemeExtension>>


See example (Checked that it works):
```
abstract class ThemeExtension<T extends ThemeExtension<T>> {
  Object get id => T;

  /// Creates a copy of this theme extension scheme with the given fields
  /// replaced by the non-null parameter values.
  ThemeExtension<T> copyWith();
}

class SomeExtension extends ThemeExtension<SomeExtension> {
  @override
  SomeExtension copyWith() => SomeExtension();
}
void main() {
  final extensions = <Object, ThemeExtension<ThemeExtension>>{};
  extensions[SomeExtension] = SomeExtension();
  
  print(extensions);
}
```

Reply:
Thanks, went with this and ignoring the linter for type specification